src/mirk.jl

```diff
@@ -0,0 +1,212 @@
+struct SimpleMIRK4{N} <: AbstractSimpleMIRK
+    nlsolve::N
+end
+SimpleMIRK4(; nlsolve = SimpleNewtonRaphson()) = SimpleMIRK4(nlsolve)
+
+struct SimpleMIRK5{N} <: AbstractSimpleMIRK
+    nlsolve::N
+end
+SimpleMIRK5(; nlsolve = SimpleNewtonRaphson()) = SimpleMIRK5(nlsolve)
+
+struct SimpleMIRK6{N} <: AbstractSimpleMIRK
+    nlsolve::N
+end
+SimpleMIRK6(; nlsolve = SimpleNewtonRaphson()) = SimpleMIRK6(nlsolve)
+
+export SimpleMIRK4
+export SimpleMIRK5
+export SimpleMIRK6
+
+alg_order(alg::SimpleMIRK4) = 4
+alg_stage(alg::SimpleMIRK4) = 3
+
+alg_order(alg::SimpleMIRK5) = 5
+alg_stage(alg::SimpleMIRK5) = 4
+
+alg_order(alg::SimpleMIRK6) = 6
+alg_stage(alg::SimpleMIRK6) = 5
+
+@concrete mutable struct SimpleMIRKCache{iip, T}
+    prob
+    alg
+    mesh
+    N
+    M
+    p
+    pt
+
+    order
+    stage
+
+    c
+    v
+    b
+    x
+
+    y
+    y0
+    residual
+    discrete_stages
+    guess
+
+    dt
+    kwargs
+end
```

Comment:
typically we have "Simple" algorithms to not have caching and other things, so that they work a bit nicer when trying to use them with StaticArrays inside GPU kernels

Reply:
So we just make the SimpleMIRK directly as:

```julia
function DiffEqBase.solve(prob::BVProblem, alg::SimpleMIRK4, dt=0.0, kwargs...)
    ......
    sol = DiffEqBase.build_solution(prob, alg, mesh, u)
end
```

we put all the nonlinear solving and collocation in this one function like what we did in https://github.com/SciML/SimpleDiffEq.jl/blob/master/src/rk4/gpurk4.jl?

Reply:
right

Reply:
I misread it partially the nonlinear solving can be offloaded to simplenonlinearsolve since that works inside kernels. Rest put it in a single function